src/librustc_resolve/lib.rs

```diff
@@ -242,20 +238,15 @@ fn resolve_struct_error<'sess, 'a>(resolver: &'sess Resolver,
                         err.span_label(span, "type variable from outer function");
                     }
                 }
-                Def::ConstParam(def_id) => {
-                    if let Some(span) = resolver.definitions.opt_span(def_id) {
-                        err.span_label(span, "const variable from outer function");
-                    }
-                }
```

Comment:
This also looks incorrectly reverted, perhaps this is the source of test failures.